extensions/src/main/java/org/pentaho/platform/web/http/api/resources/FileResource.java

```diff
@@ -491,6 +492,27 @@ public Response doGetFileOrDir( @PathParam ( "pathId" ) String pathId ) {
     }
   }
 
+  @GET
+  @Path( "/does-file-exists" )
+  @StatusCodes( {
+          @ResponseCode( code = 204, condition = "Successfully finds the file." ),
+          @ResponseCode( code = 404, condition = "Invalid Input." ),
+          @ResponseCode( code = 404, condition = "Failed to find the file." ),
+          @ResponseCode( code = 500, condition = "For any other exceptions." )
+  } )
+  public Response doesFileExists( @QueryParam ( "pathId" ) String pathId ) {
```

Comment:
you need to follow the same pattern of this class FileResource of adding a javadoc comment verbosely explaing what the rest endpoint is for `FileResource.java#doesFileExists` ?